Add permission check for summoning Empress of Light + ForceTime check for using Enchanted Sundial

moisterrific opened the pull request:
<!-- Warning: If you create a pull request and wish to remain anonymous, you are highly advised to use Tails (https://tails.boum.org/) or a fresh git environment. We will *not* be able to help with anonymization after your pull request has been created. -->

Players without `summonboss` permission should not be able to kill Prismatic Lacewing because this summon the Empress of Light. (Also we won't have to worry about other ways to get around this since EoL can ONLY be summoned if the critter was killed by players)

As for **Enchanted Sundial**, without this check, the game would try to fast forward time and TShock will keep trying to set it back to day (or night), so just use your imagination on how janky this would look lol

?????? HAVE YOU UPDATED THE CHANGELOG? ??????
yes

moisterrific committed:
Add perm check for EoL + Sundial ForceTime check

If the player does not have permission to summon bosses, they should not be able to kill Prismatic Lacewing, which summons the Empress of Light. 
Using the Enchanted Sundial while ForceTime is set to day or night (via config) will conflict with TShock's continued attempts to set it back to day or night, this makes the world appear very glitchy.

moisterrific committed:
Update CHANGELOG.md

QuiCM reviewed (dismissed):
LGTM

bartico6 commented on TShockAPI/GetDataHandlers.cs:
```diff
@@ -3201,11 +3208,20 @@ private static bool HandleSpecial(GetDataHandlerArgs args)
+				else if (TShock.Config.Settings.ForceTime != "normal")
+				{
+					TShock.Log.ConsoleDebug($"GetDataHandlers / HandleSpecial rejected enchanted sundial permission (ForceTime) { args.Player.Name}");
+					args.Player.SendErrorMessage($"You must set ForceTime to normal via config to use the Enchanted Sundial.");
```
I think this could be made more user-friendly, as this message will appear for all players, not just admins. Maybe a "Time is currently frozen on this server, and the sundial cannot be used" type message with optional information how to disable it? idk

moisterrific replied:
yea ur right that would be better for the avg player

would it make more sense to check if the player has `tshock cfg` permission and send the current msg to them only? 
its not like that info is useful for players who can use sundial but can't change the config

bartico6 replied:
Hm. This would be easier in the realm of flag-based admin tools - for instance, a SourceMod server would tell this configuration tidbit to any player with the `b` flag ("admin generic"), but for tshock,... I'm not sure what should determine adminship for the purpose of this message. Maybe access to /reload? cc @QuiCM @hakusaro

moisterrific replied:
I could change it to something like this:
```csharp
if (player has config editing permission) 
{
   "ForceTime must be set to normal to use sundial"
}
else
{
   "time is frozen so u can't use sundial"
}
```
?

moisterrific replied:
> Hm. This would be easier in the realm of flag-based admin tools - for instance, a SourceMod server would tell this configuration tidbit to any player with the `b` flag ("admin generic"), but for tshock,... I'm not sure what should determine adminship for the purpose of this message. Maybe access to /reload? cc @QuiCM @hakusaro

yea reload sounds good, sorry I thought there was some other perm related to the config lol

bartico6 reviewed (dismissed):
One small nitpick if you feel like addressing it (only if @QuiCM agrees), everything else LGTM.

moisterrific committed:
Change EoL summon to be more consistent w/ config

now this should be more in line with how other boss summons are currently handled, also made the sundial user messages better thx to quake's suggestions

moisterrific committed:
update change log again

bartico6 reviewed (dismissed):
Looks great now, no complaints :)

moisterrific committed:
Merge branch 'general-devel' into patch-29
